src/day-four/script-one.ts

```diff
@@ -52,52 +71,24 @@ function hasAdjacent(pw: number): boolean {
 	}
 	return false;
 }
-function meetsCriteria(pw: number, range: number[]): boolean {
-	const sorted = pw
-		.toString()
-		.split('')
-		.sort()
-		.map(Number);
-	if (!hasValidLength(pw)) {
-		return false;
-	}
-	if (!isInRange(pw, range)) {
-		return false;
-	}
-	if (!isNeverDecreasing(pw)) {
-		return false;
-	}
+
+function adjacentHasNoLargerGroup(pw: number): boolean {
 	if (!hasAdjacent(pw)) {
 		return false;
 	}
-	let adjacentNoLargerGroup = false;
-	const indexesOfAdjacentDigits = {};
-	for (const i in sorted) {
-		if (sorted.length > 0) {
-			const indeces: number[] = [];
-			if (sorted[i] === sorted[Number(i) + 1]) {
-				hasAdjacent = true;
-				if (indeces.indexOf(i) === -1) {
-					indeces.push(Number(i), Number(i) + 1);
-				}
-				indexesOfAdjacentDigits[sorted[i]] =
-					indexesOfAdjacentDigits[sorted[i]] === undefined
-						? [...indeces]
-						: [...indexesOfAdjacentDigits[sorted[i]], indeces[1]];
+
+	const arrOfDigits = pw.toString().split('');
+	for (const i in arrOfDigits) {
+		if (arrOfDigits.length > 0) {
+			const digit = arrOfDigits[i];
+			if (
+				digit === arrOfDigits[Number(i) + 1] &&
+				digit !== arrOfDigits[Number(i) + 2] &&
+				digit !== arrOfDigits[Number(i) - 1]
+			) {
+				return true;
 			}
+			return false;
```

Comment:
@nicolasdular What do you think about this implementation of check if an adjacent group of digits is not part of a larger group? 
[See the requirement](https://paste.pics/9c5290ea1f8eeabaa0e0300c2d0a0656)